include/core/truthTable/truth_table.hpp

```diff
@@ -82,12 +96,13 @@ namespace syrec {
 
             // pass-through functions for underlying vector
 
-            auto at(std::size_t pos) -> Value& {
+            auto operator[](std::size_t pos) -> Value& {
                 return cube[pos];
             }
 
             auto operator[](std::size_t pos) const -> Value {
-                return cube[pos];
+                const std::vector<Value>& x = cube;
+                return x[pos];
             }
```

Comment:
This is not a proper definition of the const version of `operator[]`. It should be
```c++
auto operator[](std::size_t pos) -> const Value& {
  return cube[pos];
}
```
Especially, it should not return a value, but a const reference and there is no need to create a temporary.

Reply:
I had tried this earlier, however, I was getting the following error:

Functions that differ only in their return type cannot be overloaded

What do you suggest I do?

Reply:
I just edited the comment above. The suggestion was missing a const.
That way it should work because it is a different signature.

Reply:
done 👍🏼